### PR TITLE
show modal with impacted proposals when changing voting duration

### LINKED ISSUE #145: [Task] Show affected proposals in a popup when changing voting duration
### Background

Right now the proposals affected by changing voting duration, are displayed in a table below the input field as you type. To reduce disturbance in the user experience, there should rather be a popup showing this information when you click the submit button.

### User Story

As an admin
I want a popup when clicking the submit button after changing voting duration
So that I can see which proposals that will expire, or which will have their expiry date prolonged

as described in #119 

### Acceptance Criteria

- [x] The current list of affected proposals, that is updated as you type, should be removed
- [x] A popup like in the screenshot below should be shown when clicking the submit button ( and only if there are proposals that will be affected by the voting duration change )
- [x] Expired proposals that become active when extending duration days, should also be visible
- [x] Show pending proposals that will follow the new voting duration ( without expiring or becoming active )

### PR DESCRIPTION
**Pull request by petersalomonsen**
- Removes the current list of affected proposals, that is updated as you type
- Adds a when clicking the "Submit request" button ( and only if there are proposals that will be affected by the voting duration change )
- Shows expired proposals that become active when extending duration days
- Shows pending proposals that will follow the new voting duration ( without expiring or becoming active )

https://github.com/user-attachments/assets/f858e549-b293-4122-9baf-ca3157c32dd3

resolves #145

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@petersalomonsen Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/148&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/148&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/petersalomonsen)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/petersalomonsen)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: instances/treasury-devdao.near/widget/lib/modal.jsx
Comment (Megha-Dev-19) [inline instances/treasury-devdao.near/widget/lib/modal.jsx:109]: just FYI, we also have this [library](https://www.uiisnear.xyz/dialog)

**Review by Megha-Dev-19 — APPROVED** [review]
LGTM

**Comment by Megha-Dev-19** [comment]
@race-of-sloths score 5

**Comment by ori-near** [comment]
> @ori-near @FREZZZBE See the video above. Regarding the list of impacted proposals. Do you think it should show all pending proposals, or just the ones that will move from pending->expired or expired->pending? Right now it only shows those that will change expired/pending state, while there is still a count ( in the first bullet ) of how many proposals that will follow the new policy without changing state.

Thank you @petersalomonsen ... we're getting super close, just have a couple of formatting fixes. Can you please bold "Pending Requests" and "Expired Requests" and also remove "Impacted Requests" bullet and create a Summary of Changes  header in a separate line under underneath. Also would be great if we can add the warning icon. 

See reference:

Screenshot 2024-12-03 at 17 25 36

**Comment by petersalomonsen** [comment]
> > @ori-near @FREZZZBE See the video above. Regarding the list of impacted proposals. Do you think it should show all pending proposals, or just the ones that will move from pending->expired or expired->pending? Right now it only shows those that will change expired/pending state, while there is still a count ( in the first bullet ) of how many proposals that will follow the new policy without changing state.
> 
> Thank you @petersalomonsen ... we're getting super close, just have a couple of formatting fixes. Can you please bold "Pending Requests" and "Expired Requests" and also remove "Impacted Requests" bullet and create a Summary of Changes header in a separate line under underneath. Also would be great if we can add the warning icon.
> 
> See reference:
> 
> Screenshot 2024-12-03 at 17 25 36

I've made the adjustments @ori-near @FREZZZBE Now it looks like this:

image

I also notice that the table has a different look and feel in your reference image. Should I look into that as part of this PR, or should we follow up that separately and create a standard table style across the portal?

**Pull request merged**
